mysql-test/suite/innodb/t/optimizer_temporary_table_2.test

```diff
@@ -0,0 +1,57 @@
+--echo #
+--echo # Test for PS-8772: "Stall of instance after INSERT (...) SELECT".
+--echo #
+
+--echo # Check that InnoDB buffer pool size is set to minimum/5Mb and thus is
+--echo # small enough to trigger original problem. If it ever changes the size
+--echo # of source table below needs to be changed proportionately.
+--let $innodb_buffer_pool_size = `SELECT @@global.innodb_buffer_pool_size`
+--let $assert_text = InnoDB Buffer Pool has minimal size (5Mb).
+--let $assert_cond = $innodb_buffer_pool_size = 5242880
+--source include/assert.inc
+
+CREATE TABLE source (id int primary key auto_increment, a varchar(100), b varchar(100), c varchar(100), d varchar(100), e varchar(100), hdl_source_id int, hdl_created_date datetime);
+CREATE TABLE dest LIKE source;
+
+--echo # Populate source table with 64K records.
+--disable_query_log
+INSERT INTO source (a, b, c, d, e, hdl_source_id, hdl_created_date) values ('aaaaaaaaa', 'bbbbbbbbbb', 'cccccccccccccccc', 'dddddddddddddddddddddddddddd', 'eeeeeeeeeeeeeeeeee', 1, NOW());
+--let $i= 0
+while ($i < 16)
+{
+  INSERT INTO source (a, b, c, d, e, hdl_source_id, hdl_created_date) SELECT a, b, c, d, e, hdl_source_id + 1, NOW() FROM source;
```

Comment:
Just a suggestion here.
Percona Server has the `SEQUENCE_TABLE()` table-level function (like `JSON_TABLE()`).
https://docs.percona.com/percona-server/8.0/flexibility/sequence_table.html
Basically you can do something like
```sql
INSERT INTO source (a, b, c, d, e, hdl_source_id, hdl_created_date)
  SELECT 'aaaaaaaaa', 'bbbbbbbbbb', 'cccccccccccccccc', 'dddddddddddddddddddddddddddd', 'eeeeeeeeeeeeeeeeee', value, NOW()
    FROM SEQUENCE_TABLE(65536) AS tt;
```

Reply:
Done.